File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_a = 0
        i = 0
        j = len(height)-1
        while i < j:
            max_a = max(max_a,min(height[i],height[j]) * abs(j-i))
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        
        return max_a

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int area, max_area = 0, left = 0, right = height.length-1;
        while (left<right) {
            area = Math.min(height[left], height[right]) * (right - left);
            max_area = Math.max(area, max_area);

            if(height[left] < height[right]){
                left++;
            }else{
                right--;
            }
        }

        return max_area;
    }
}